ibm/resource_ibm_dns_reverse_record.go

```diff
@@ -0,0 +1,147 @@
+package ibm
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/softlayer/softlayer-go/datatypes"
+	"github.com/softlayer/softlayer-go/services"
+	"github.com/softlayer/softlayer-go/sl"
+)
+
+func resourceIBMDNSREVERSERecord() *schema.Resource {
+	return &schema.Resource{
+		Exists:   resourceIBMDNSREVERSERecordExists,
+		Create:   resourceIBMDNSREVERSERecordCreate,
+		Read:     resourceIBMDNSREVERSERecordRead,
+		Update:   resourceIBMDNSREVERSERecordUpdate,
+		Delete:   resourceIBMDNSREVERSERecordDelete,
+		Importer: &schema.ResourceImporter{},
+		Schema: map[string]*schema.Schema{
+			"dns_ipaddress": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"dns_hostname": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"dns_ttl": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				DefaultFunc: func() (interface{}, error) {
+					return 604800, nil
+				},
+			},
+		},
+	}
+}
+
+//  Creates DNS Domain Reverse Record
+//  https://sldn.softlayer.com/reference/services/SoftLayer_Dns_Domain/CreatePtrRecord
+func resourceIBMDNSREVERSERecordCreate(d *schema.ResourceData, meta interface{}) error {
+	sess := meta.(ClientSession).SoftLayerSession()
+	service := services.GetDnsDomainService(sess.SetRetries(0))
+	Data := sl.String(d.Get("dns_hostname").(string))
+	Ttl := sl.Int(d.Get("dns_ttl").(int))
+	Ipaddress := sl.String(d.Get("dns_ipaddress").(string))
+	var err error
```

Comment:
No need to initialize err, record varaibales..Directly you can use record, err:= ......